src/server/transaction.cc

```diff
@@ -814,7 +814,7 @@ bool Transaction::ScheduleUniqueShard(EngineShard* shard) {
   DCHECK_EQ(0, sd.local_mask & KEYLOCK_ACQUIRED);
   bool lock_acquired = shard->db_slice().Acquire(mode, lock_args);
   sd.local_mask |= KEYLOCK_ACQUIRED;
-  DCHECK(!lock_acquired);  // Because CheckLock above failed.
+  // DCHECK(!lock_acquired);  // Because CheckLock above failed.
```

Comment:
This needs a careful check: If we failed to run quickly due to the shard being locked, then this lock acquisition doesn't fail. So I commented the DCHECK out... But does the logic change in this case?